cdap-notifications-api/src/main/java/co/cask/cdap/notifications/feeds/NotificationFeedManager.java

```diff
@@ -31,31 +33,32 @@
    * @throws NotificationFeedException if the feed has an incorrect structure;
    * for example, if it is missing a name, a namespace or a category
    */
-  boolean createFeed(NotificationFeed feed) throws NotificationFeedException;
+  boolean createFeed(Id.NotificationFeed feed) throws NotificationFeedException;
```

Comment:
It doesn't seem strange to me at all: you want to perform an action, and it returns false if that action did not work. Then the javadoc tells you in which case it did not work. It kind of is the same as the ``Collection.add`` method signature (http://docs.oracle.com/javase/7/docs/api/java/util/Collection.html#add(E)). It returns true if the collection as changed, just like we do here, and false if not, again like we do here.

I feel like a lot of classes follow various contracts, and in the end not any one is better than the other.